Draft JOSS paper (#451)
easystats/easystats: paper/paper.md

# A Harmonized and Integrated Workflow

A key design principle of the **easystats** ecosystem is the harmonization and integration of different packages into a simple, sequential workflow. The typical workflow for a statistical analysis using `{easystats}` starts with importing data and bringing the data into shape for the next step—fitting a model—and then sequentially using different functions to obtain a comprehensive understanding of the model. This can include checking the model's parameters, performance metrics, specific effect sizes [e.g., @ben2023phi], and statistical outliers [@theriault2024check], as well as obtaining publication-ready figures and written summaries of the results.

> Let's demonstrate this with an example, where the user starts by preparing some data and then fits a logistic mixed effects model:

Reviewer: ```suggestion
Let's demonstrate this with an example, where the user wishes to fit a logistic mixed effects model for a dataset:
```

``` r
# we don't load each package individually,
# but rather the entire ecosystem
library(easystats)
data(coffee_data, package = "modelbased")

# dichotomize outcome variable

> coffee_data$alertness <- categorize(coffee_data$alertness, lowest = 0)

Reviewer: I am not a statistician, but isn't dichotomizing a continuous variable and throwing away valuable variance considered a bad practice? Can't we use a dataset with a naturally occurring dichotomous response (like yes/no, on/off, up/down, etc.) to showcase this model?

# rename variable
coffee_data <- data_rename(coffee_data, select = c(treatment = "coffee"))

# fit mixed model
model <- glmmTMB::glmmTMB(
  alertness ~ treatment + (1 | ID),
  data = coffee_data,
  family = binomial()
)
```

The `model` object can then be passed to functions from different **easystats** packages. For instance, the user can get a summary of the model parameters using the `{parameters}` package:

``` r
model_parameters(model)
#> # Fixed Effects
#> 
#> Parameter           | Log-Odds |   SE |        95% CI |     z |     p
#> ---------------------------------------------------------------------
#> (Intercept)         |     0.13 | 0.26 | [-0.37, 0.64] |  0.52 | 0.606
#> treatment [control] |    -0.27 | 0.37 | [-0.99, 0.45] | -0.73 | 0.466
#> 
#> # Random Effects
#> 
#> Parameter          | Coefficient |           95% CI
#> ---------------------------------------------------
#> SD (Intercept: ID) |        0.05 | [0.00, 9.86e+28]
```

Then, the performance of the model can be assessed with the `{performance}` package:

``` r
model_performance(
  model,

>   metrics = c("AIC", "BIC", "R2")

Reviewer: Why select only particular metrics, instead of showcasing everything it returns by default?

Author: Because of line-overflow

)
#> # Indices of model performance
#> 
#> AIC   |   BIC | R2 (cond.) | R2 (marg.)
#> ---------------------------------------
#> 171.8 | 180.2 |      0.006 |      0.005
```

The results can be visualized using the `{see}` package by, for example, plotting the model's predictions from `{modelbased}` (Figure 1):

``` r
predictions <- estimate_means(model, "treatment")
plot(predictions) + theme_modern(show.ticks = TRUE) # add nice theme
```

\begin{figure}
\includegraphics[width=1\linewidth]{paper_files/figure-latex/fig1-1} \caption{Predicted probability of alertness by treatment group.}\label{fig:fig1}
\end{figure}

Finally, a full report of the analysis can be generated with the `{report}` package:

> 

Reviewer: What's with all the blank lines?

#> R2 = 6.35e-03) and the part related to the fixed effects alone
#> (marginal R2) is of 5.44e-03. The model's intercept, corresponding
#> to treatment = coffee, is at 0.13 (95% CI [-0.37, 0.64], p = 0.606).
#> Within this model:
#> 
#>   - The effect of treatment [control] is statistically non-significant
#> and negative (beta = -0.27, 95% CI [-0.99, 0.45], p = 0.466)
#>   - The effect of treatment [control] is statistically non-significant
#> and negative (beta = -0.27, 95% CI [-0.99, 0.45], p = 0.466)
#>   - The effect of treatment [control] is statistically non-significant
#> and negative (beta = -0.27, 95% CI [-0.99, 0.45], p = 0.466)
#>   - The effect of treatment [control] is statistically non-significant
#> and negative (beta = -0.27, 95% CI [-0.99, 0.45], p = 0.466)
#>   - The effect of treatment [control] is statistically non-significant
#> and negative (beta = -0.27, 95% CI [-0.99, 0.45], p = 0.466)
#> 
#> Standardized parameters were obtained by fitting the model on a
#> standardized version of the dataset. 95% Confidence Intervals (CIs)
#> and p-values were computed using a Wald z-distribution
#> approximation.
```

> The `model_dashboard()` function generates an interactive model summary, integrating several **easystats** steps into one command. Since this paper is a static document, we cannot show a live demonstration, but Figure 2 provides a visual example.

Reviewer: ```suggestion
The `model_dashboard()` function generates an interactive model summary, integrating several **easystats** steps into a single command. Since this paper is a static document, Figure 2 provides a screenshot of the interactive report.
```
